main/src/mill/main/MainModule.scala

```diff
@@ -173,4 +179,34 @@ trait MainModule extends mill.Module{
       }
     }
   }
+
+  /**
+    * Deletes the given targets from the out directory. Providing no targets will clean everything.
+    */
+  def clean(targets: String*) = mill.T.command {
+    val result = Try {
+      if (ammonite.ops.ls(ammonite.ops.pwd).contains(ammonite.ops.pwd / BuildFile)) {
+        targets match {
+          case Nil =>
+            remove(List(ammonite.ops.pwd / OutDir))
+          case _ =>
+            remove(targets
+              .map(_.replace(".", "/"))
+              .map(target => Path(target, ammonite.ops.pwd / OutDir))
+              .toList)
+        }
+      } else {
+        Result.Failure(s"Cannot find $BuildFile")
+      }
+    }
+    result match {
+      case Success(r) => r
+      case Failure(error) => Result.Exception(error, new Result.OuterStack(error.getStackTrace))
+    }
+  }
+
+  private def remove(paths: List[Path]): Result[String] = {
+    paths.filter(ammonite.ops.exists).foreach(ammonite.ops.rm)
```

Comment:
You don't need the `filter`; `rm` already just ignores a path if there' nothing to remove

Reply:
Oh nice.